lib/ansible/modules/cloud/vmware/vmware_drs.py

```diff
@@ -0,0 +1,279 @@
+        self.create_rule_spec()
+
+    def create_rule_spec(self):
+        """
+        Function to create DRS rule
+        Returns: Changed status and result
+
+        """
+        if self.affinity_rule:
+            rule = vim.cluster.AffinityRuleSpec(vm=self.vm_obj_list,
+                                                enabled=self.enabled,
+                                                mandatory=self.mandatory,
+                                                name=self.rule_name)
+        else:
+            rule = vim.cluster.AntiAffinityRuleSpec(vm=self.vm_obj_list,
+                                                    enabled=self.enabled,
+                                                    mandatory=self.mandatory,
+                                                    name=self.rule_name)
+        rule_spec = vim.cluster.RuleSpec(info=rule, operation='add')
+        config_spec = vim.cluster.ConfigSpecEx(rulesSpec=[rule_spec])
+        task = self.cluster_obj.ReconfigureEx(config_spec, modify=True)
+        changed, result = wait_for_task(task)
+        if changed:
+            self.module.exit_json(changed=changed, results=result)
+        else:
+            self.module.fail_json(msg="Failed to create DRS Rule %s" % self.rule_name, details=result)
+
+    # Delete
+    def delete(self):
+        """
+        Function to delete DRS rule using name
+        Returns: True if delete successful, else false
+
+        """
+        ret = False
+        rule_key = self.get_rule_key_by_name(rule_name=self.rule_name)
+        if rule_key:
+            rule_key = int(rule_key)
+            rule_spec = vim.cluster.RuleSpec(removeKey=rule_key, operation='remove')
+            config_spec = vim.cluster.ConfigSpecEx(rulesSpec=[rule_spec])
+            task = self.cluster_obj.ReconfigureEx(config_spec, modify=True)
+            changed, result = wait_for_task(task)
+            if changed:
+                self.module.exit_json(changed=changed, results=result)
+            else:
+                self.module.fail_json(msg="Failed to delete DRS rule %s" % self.rule_name, details=result)
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(
+        state=dict(type='str', default='present', choices=['absent', 'present']),
+        vms=dict(type='list', default=[]),
+        cluster=dict(type='str', required=True),
+        drs_rule_name=dict(type='str', required=True),
+        enabled=dict(type='bool', default=False),
+        mandatory=dict(type='bool', default=False),
+        affinity_rule=dict(type='bool', default=True),
+    )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+
+    results = dict(failed=False, changed=False)
+
+    state = module.params['state']
+    vm_drs = VmwareDrs(module)
+
+    if state == 'present':
+        # Add Rule
+        if not module.check_mode:
```

Comment:
`if not module.check_mode` → `if module.check_mode`

Reply:
ok.